airflow/models/baseoperator.py

```diff
@@ -1838,6 +1839,38 @@ def cross_downstream(
         task.set_downstream(to_tasks)
 
 
+def chain_sequential(*elements: DependencyMixin | Sequence[DependencyMixin]):
```

Comment:
if you have better suggestion of name @uranusjr ... i'll take it

`chain_tasks`

`chain_sequential` is both right and wrong, as would be `chain_parallel`

`chain_the_other_way` 🤦

Reply:
~~`chain_correctly`~~

Maybe something to describe the fact each “group” is joined? One that comes immediately to me is `chain_reduced` since this is very much like how dynamic task mapping works when each mapped task is automatically joined (and looking back these really should have been consistent with each other). If I can design this from scratch I’d call _this_ `chain` and the other `chain_branched` but now we are stuck to coming up with the opposite of branching.

Reply:
As this seemed similar to python's `zip_longest` even though it's not using None to compensate for unequal mapping(it can actually be argued), `chain_longest` looks good IMO.

Reply:
thanks for the suggestions.

well, here's the argument for sequential.  "sequential" here means, each element provided will be arrowed to the next, in sequence, whether list or otherwise.  maybe `chain_linear` would be another option, perhaps better cus it doesn't invoke the concept of sequences, or sequential vs parallel.
wdyt?

longest doesn't quite feel right because as you mention, we don't "fill out" anything to match the longest.

i do see the connection with reduce... because effectively we're reducing the elements with `>>`. maybe i would pick `chain_reduce` over `chain_reduced`.  the thing that steers me away though is the fact that it invokes the `reduce` notion which makes it sound like it's doing something related to mapping.

Reply:
i think the orig one should probably be called `chain_zip`

Reply:
i guess chain reduce grows on me a bit.  unless `chain_linear` seemed better to you, maybe we can add this as `chain_reduce` and rename the other to `chain_zip`.

Reply:
Even having `chain_zip` and `chain_linear` does not seem too weird to me either.

Reply:
Yeah maybe I'll go with those too tomorrow....

Reply:
ok -- i renamed to linear and i will add the "chain -> chain_zip" rename in a separate PR